src/platform/sdl/main.c

```diff
@@ -149,6 +154,15 @@ int main(int argc, char** argv) {
 	renderer.player.bindings = &renderer.core->inputMap;
 	mSDLInitBindingsGBA(&renderer.core->inputMap);
 	mSDLInitEvents(&renderer.events);
+
+#ifdef USE_PLEDGE
+	if (!mPledgeBroad(&args)) {
```

Comment:
Is there any reason this isn't done immediately after generating the `args` struct? There's a lot of stuff you're not freeing or deiniting here.

Reply:
Joystick initialization calls some USB ioctls that are not allowed by any pledge, so this is as early as the pledge call can go.

Reply:
I gave this another try.